src/bot.js

```diff
@@ -1,38 +1,186 @@
+    link += data.url.substring(30);
+    await channel.send(link).catch(console.log);
+  }
+  else await channel.send(data.url).catch(console.log);
 
-    embed.setTitle(data.title)
-         .setImage(data.hdurl)
-         .setDescription(data.explanation)
-         .setColor('PURPLE')
-         .setFooter('APOD for ' + data.date + ' • Photo by ' + data.copyright);
+  let footer = 'APOD for ' + data.date;
+  if (data.copyright) footer += ` • ${data.media_type} by ` + data.copyright;
 
-    await channel.send(embed).catch(console.log);
-  });
+  embed
+    .setDescription(data.explanation)
+    .setColor('PURPLE')
+    .setFooter(footer);
+
+  await channel.send(embed).catch(console.log);
+}
+
+/**
+ * This function is called when a connection between the bot and Discord's
+ * websocket has been established.
+ */
+function onReady() {
+  // this = Client
+  const { user } = this;
+  // if the bot is not logged in then user is null
+  if (user !== undefined) {
+    console.log(`Ready as: ${user.tag}`);
+    console.log(`Client ID: ${user.id}`);
+  }
+  client.user.setActivity('apod.help');
+  // TODO: set jobs for all lines in jobs.csv
+}
+
+/**
+ * Handles incoming messages that the Discord bot can see
+ * @param {Message} msg
+ * @returns {Promise<void>}
+ */
+async function onMessage(msg) {
+  // this = Client
+  const { user } = this;
+
+  // this will allow us to ignore case-sensitivity when talking to the bot
+  const cmd = msg.content.toLowerCase();
+
+  if (cmd.substring(0, 5) === 'apod.') {
+    let method = '';
+    let args = [];
+    if (cmd.indexOf(' ') === -1) method = cmd.substring(5)
+    else {
+      method = cmd.substring(5, cmd.indexOf(' '));
+      args = cmd.substring(cmd.indexOf(' ')+1).split(' ');
+    }
+
+    switch (method) {
+      case 'get_invite':
+        const botURL = 'https://discord.com/oauth2/authorize?client_id=';
+        msg.reply(botURL+user.id+'&scope=bot');
+        break;
+      case 'set':
+        // TODO: need to add check that channel is text channel and that bot has permission to post in it
+
+        // if channel provided is in messagers' guild, then proceed with scheduling job
+        if (msg.guild.channels.cache.get(args[0])) {
+          if (ns.cancelJob(args[0])) console.log('job already existed'); // remove job from csv
+          let h = args[1].substring(0,2);
+          let m = args[1].substring(2);
+          let cron = `${+m} ${+h} * * *`;
+          ns.scheduleJob(args[0], cron, async () => { await sendAPOD(args[0]) });
+          // TODO: add job to csv
+          await msg.reply(`Job added to ${args[0]} for ${args[1]} daily`);
+        }
+        else await msg.reply('Error: Invalid Channel ID');
+        break;
+      case 'cancel':
+        // TODO: need to add check that channel is text channel and that bot has permission to post in it
+
+        // if channel provided is in guild, try to cancel job
+        if (msg.guild.channels.cache.get(args[0]) && ns.cancelJob(args[0])) {
```

Comment:
This if statement is quite long, perhaps put the get function call in a constant?
```js
const channel = await msg.guild.channels.fetch(args[0]);
if (channel && ns.cancelJob(args[0])) {
// ...
}

Reply:
https://github.com/mazacon/apod-bot/pull/3/commits/a2b2b5b76ebbe7d215f443cc18f37ed18e2d9061